### ISSUE TITLE
Update S1104: Rule should ignore DependencyProperty fields

### ISSUE THREAD

**Issue by alanwol** [SonarSource/sonar-dotnet#2141]
### Description

False positive for rule S1104 for dependency properties in WPF code.

https://docs.microsoft.com/en-us/dotnet/framework/wpf/advanced/dependency-properties-overview

### Repro steps

public static DependencyProperty HeaderProperty = DependencyProperty.Register("Header", typeof(string), typeof(MyControl), new FrameworkPropertyMetadata(null));

### Related information

This is probably related to: https://github.com/SonarSource/sonar-dotnet/issues/1317

SonarLint version: 4.7.0.3891
SonarQube c# plugin: 7.9.1 (build 7622) 
Visual Studio 15.8.2
SonarQube scanner 4.3.2

**Comment by Evangelink** [comment]
Hi @alanwol,

Thanks for the feedback.

If you mark your field as `readonly` the issue won't be raised. For info, this is what the documentation is doing:

```csharp
public static readonly DependencyProperty IsSpinningProperty = 
    DependencyProperty.Register(
    "IsSpinning", typeof(Boolean),
    typeof(MyCode)
    );
```

Cheers,
Amaury
